app/test/unit/git/status-test.ts

```diff
@@ -19,110 +20,144 @@ const _temp = temp.track()
 
-    it(
-      'Handles at least 10k untracked files without failing',
-      async () => {
-        const numFiles = 10000
-        const basePath = repository!.path
+        await FSE.writeFile(path.join(repo.path, 'foo'), 'foo\n')
 
-        await mkdir(basePath)
+        await GitProcess.exec(['add', 'foo'], repo.path)
+        await GitProcess.exec(['commit', '-m', 'Initial commit'], repo.path)
+        await GitProcess.exec(['mv', 'foo', 'bar'], repo.path)
 
-        // create a lot of files
-        const promises = []
-        for (let i = 0; i < numFiles; i++) {
-          promises.push(
-            FSE.writeFile(path.join(basePath, `test-file-${i}`), 'Hey there\n')
-          )
-        }
-        await Promise.all(promises)
+        const status = await getStatusOrThrow(repo)
+        const files = status.workingDirectory.files
 
-        const status = await getStatusOrThrow(repository!)
+        expect(files.length).to.equal(1)
+        expect(files[0].status).to.equal(AppFileStatus.Renamed)
+        expect(files[0].oldPath).to.equal('foo')
+        expect(files[0].path).to.equal('bar')
+      })
+
+      it('reflects copies', async () => {
+        const testRepoPath = await setupFixtureRepository(
+          'copy-detection-status'
+        )
+        repository = new Repository(testRepoPath, -1, null, false)
+
+        // Git 2.18 now uses a new config value to handle detecting copies, so
+        // users who have this enabled will see this. For reference, Desktop does
+        // not enable this by default.
+        await GitProcess.exec(
+          ['config', '--local', 'status.renames', 'copies'],
+          repository.path
+        )
+
+        await GitProcess.exec(['add', '.'], repository.path)
+
+        const status = await getStatusOrThrow(repository)
         const files = status.workingDirectory.files
-        expect(files.length).to.equal(numFiles)
-      },
-      // needs a little extra time on CI
-      25000
-    )
-
-    it('returns null for directory without a .git directory', async () => {
-      repository = setupEmptyDirectory()
-      const status = await getStatus(repository)
-      expect(status).is.null
+
+        expect(files.length).to.equal(2)
+
+        expect(files[0].status).to.equal(AppFileStatus.Modified)
+        expect(files[0].oldPath).to.be.undefined
+        expect(files[0].path).to.equal('CONTRIBUTING.md')
+
+        expect(files[1].status).to.equal(AppFileStatus.Copied)
+        expect(files[1].oldPath).to.equal('CONTRIBUTING.md')
+        expect(files[1].path).to.equal('docs/OVERVIEW.md')
+      })
+
+      it(
+        'Handles at least 10k untracked files without failing',
+        async () => {
+          const numFiles = 10000
+          const basePath = repository!.path
+
+          await mkdir(basePath)
+
+          // create a lot of files
+          const promises = []
+          for (let i = 0; i < numFiles; i++) {
+            promises.push(
+              FSE.writeFile(
```

Comment:
This can be simplified by using FSE.writeFileSync. Given that this is a test, it's fine if it's not async.

Reply:
I didn't write this, I only indented it. I think its worth seeing if this runs faster on CI if done synchronously, but I think I'm leaning towards that being a separate PR/issue. @shiftkey do you have an opinion on this?

Reply:
Personally I'd just do a dumb `for` loop and `await` on the inside rather than create all these promises. Or use the `Sync` API.

Reply:
I tried this out and it slowed the test execution by ~30% locally (~+.4 seconds) and on CircleCI (~+8 seconds). Let's not remove the promises approach.